### PR TITLE
Infinitas Shamir: Adds 2 packtests

### PR DESCRIPTION
**Pull request by EpyonProjects**
- The most basic, does it pick up liquids without breaking
- A more complicated check for data breaking between versions for unsmooshing

@misode, is there a way to build off of a previous test?  The second one is just the first one with extra steps

### CODE REVIEW

**Comment by misode** [comment]
> is there a way to build off of a previous test? The second one is just the first one with extra steps

You can either delete the first test, since all those things are also covered by the second test. Or you can keep the duplicate test logic. There is no way to reuse a previous test, but you can manually create the success state of the previous test with NBT or a different structure template.

File: gm4_metallurgy/data/gm4_infinitas_shamir/tests/check_unsmoosh_after_use.mcfunction [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+dummy @s drop
+tp @e[type=item,distance=..6,sort=nearest,limit=1,nbt={Item:{id:"minecraft:bucket",Count:1b,tag:{gm4_metallurgy:{active_shamir:"infinitas"}}}}] ~1.5 ~2.5 ~1.5
+
+await delay 1s
+
+setblock ~1 ~4 ~1 redstone_block
+
+await entity @e[type=item,distance=..6,nbt={Item:{id:"minecraft:player_head"}}]
+execute unless data entity @e[type=item,limit=1,nbt={Item:{id:"minecraft:player_head"}}] Item.tag.gm4_metallurgy.stored_shamir run fail "Shamir did not unsmoosh properly"
+execute unless data entity @e[type=item,limit=1,nbt={Item:{id:"minecraft:player_head"}}] Item.tag.display.Lore[2] run fail "Shamir did not unsmoosh properly"
```
Comment (misode) [inline gm4_metallurgy/data/gm4_infinitas_shamir/tests/check_unsmoosh_after_use.mcfunction:41]: Unless you have a more specific failure message it's probably best to use `assert data` here, which gives better messages:

```mcfunction
assert data entity @e[type=item,limit=1,nbt={Item:{id:"minecraft:player_head"}}] Item.tag.gm4_metallurgy.stored_shamir
```

Comment (misode) [reply gm4_metallurgy/data/gm4_infinitas_shamir/tests/check_unsmoosh_after_use.mcfunction]: Indeed, `assert` is basically `execute unless <condition> run fail <nice error message>`

**Comment by EpyonProjects** [comment]
I'm not crazy, right??

**Comment by EpyonProjects** [comment]
I've been thinking about the duplicated code more.  I could probably just have an assert in the middle of the longer one to check what the first does

**Pull request merged**